pkg/util/mount/mount_linux.go

```diff
@@ -185,12 +190,17 @@ func (mounter *Mounter) PathIsDevice(pathname string) (bool, error) {
 }
 
 func exclusiveOpenFailsOnDevice(pathname string) (bool, error) {
-	if isDevice, err := pathIsDevice(pathname); !isDevice {
+	isDevice, err := pathIsDevice(pathname)
+	if err != nil {
 		return false, fmt.Errorf(
 			"PathIsDevice failed for path %q: %v",
 			pathname,
 			err)
 	}
+	if !isDevice {
+		glog.Errorf("Path %q is not refering to a device.", pathname)
+		return false, nil
```

Comment:
Yes. If the path is not a device (this might happen e.g., when reconstructing state from directory, no device path can be recovered, and the state will have an empty device path), there is no need to check whether device is in use or not since it is not a device at all. So the function call will return false without error. 
If it returns error in case of passing path is not a device path, operation_executor will return with error without marking volume as unmounted.